File: Common/InMotionGIT.Common/Extensions/DataCommandExtensions.cs
using System;
using System.Collections.Generic;
using System.Data;

namespace InMotionGIT.Common.Extensions
{

    /// <summary>
    /// Extension methods for the Services.Contracts.DataParameter data type
    /// </summary>
    public static class DataCommandExtensions
    {

        public static void AddParameter(ref Services.Contracts.DataCommand command, string name, DbType type, int size, bool isNull, object value, ParameterDirection direction)
        {
            if (command.Parameters.IsEmpty())
            {
                command.Parameters = new List<Services.Contracts.DataParameter>();
            }
            command.Parameters.Add(new Services.Contracts.DataParameter() { Name = name, Type = type, Size = size, IsNull = isNull, Value = value, Direction = direction });
        }

        public static void AddParameter(ref Services.Contracts.DataCommand command, string name, DbType type, int size, bool isNull, object value)
        {
            if (command.Parameters.IsEmpty())
            {
                command.Parameters = new List<Services.Contracts.DataParameter>();
            }

            command.Parameters.Add(new Services.Contracts.DataParameter() { Name = name, Type = type, Size = size, IsNull = isNull, Value = value, Direction = ParameterDirection.Input });
        }

        public static Services.Contracts.DataParameter AddParameter(ref Services.Contracts.DataCommand command, string name)
        {
            Services.Contracts.DataParameter result = null;

            foreach (Services.Contracts.DataParameter parameter in command.Parameters)
            {
                if (string.Equals(parameter.Name, name, StringComparison.CurrentCultureIgnoreCase))
                {
                    result = parameter;
                    break;
                }
            }

            return result;
        }

    }

}

File: Common/InMotionGIT.Common - Copy/Extensions/DataCommandExtensions.vb
Imports System.Runtime.CompilerServices

Namespace Extensions

    ''' <summary>
    ''' Extension methods for the Services.Contracts.DataParameter data type
    ''' </summary>
    Public Module DataCommandExtensions

        <Extension()>
        Public Sub AddParameter(ByRef command As Services.Contracts.DataCommand, name As String, type As DbType, size As Integer, isNull As Boolean, value As Object, direction As ParameterDirection)
            If command.Parameters.IsEmpty Then
                command.Parameters = New List(Of Services.Contracts.DataParameter)
            End If
            command.Parameters.Add(New Services.Contracts.DataParameter With {.Name = name, .Type = type, .Size = size, .IsNull = isNull, .Value = value, .Direction = direction})
        End Sub

        <Extension()>
        Public Sub AddParameter(ByRef command As Services.Contracts.DataCommand, name As String, type As DbType, size As Integer, isNull As Boolean, value As Object)
            If command.Parameters.IsEmpty Then
                command.Parameters = New List(Of Services.Contracts.DataParameter)
            End If

            command.Parameters.Add(New Services.Contracts.DataParameter With {.Name = name, .Type = type, .Size = size, .IsNull = isNull, .Value = value, .Direction = ParameterDirection.Input})
        End Sub

        <Extension()>
        Public Function AddParameter(ByRef command As Services.Contracts.DataCommand, name As String) As Services.Contracts.DataParameter
            Dim result As Services.Contracts.DataParameter = Nothing

            For Each parameter As Services.Contracts.DataParameter In command.Parameters
                If String.Equals(parameter.Name, name, StringComparison.CurrentCultureIgnoreCase) Then
                    result = parameter
                    Exit For
                End If
            Next

            Return result
        End Function

    End Module

End Namespace